Application.hs

```diff
@@ -162,22 +165,18 @@ develMain = develMainHelper getApplicationDev
 -- | The @main@ function for an executable running this site.
 appMain :: IO ()
 appMain = do
-    -- Get the settings from all relevant sources
-    settings <- withEnv $ loadAppSettingsArgs
-        -- fall back to compile-time values, set to [] to require values at runtime
-        [configSettingsYmlValue]
-
-        -- allow environment variables to override
-        useEnv
-
-    -- Generate the foundation from the settings
+    settings <- getAppSettings
```

Comment:
The default behavior (provided by `loadAppSettingsArgs` is to accept a filename. A feature I use [here](https://github.com/thoughtbot/carnival/pull/250/files#diff-0405cbeab00d5d74e15d255c36022ee4R8). It might also be used implicitly by `yesod test`, I'm not sure. Either way I think we need to be careful about removing it.